File: 0062 - Unique Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        std::vector<std::vector<int>> arr(m, std::vector<int>(n, 1));
        
        if(m == 1 || n == 1)
            return 1;

        for(int i = 1; i < m; i++){
            for(int j = 1; j < n; j++){
                arr[i][j] = arr[i - 1][j] + arr[i][j - 1];
            }
        }
        return arr[m - 1][n - 1];
    }
};

File: 0062 - Unique Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        if m == n == 1:
            return 1
        grid = [[1 for _ in range(m)] for _ in range(n)]
        grid[0][0] = 0
        for i in range(1, m):
            for j in range(1, n):
                grid[j][i] = grid[j][i - 1] + grid[j - 1][i]
        return grid[n - 1][m - 1]